draft-ietf-httpbis-priority.md

```diff
@@ -587,6 +588,40 @@ end-to-end rather than hop-by-hop.
 It should also be noted that the use of a header field carrying a textual value
 makes the prioritization scheme extensible; see the discussion below.
 
+# Security Considerations
+
+[CVE-2019-9513] aka "Resource Loop", is a DoS attack based on manipulation of
+the HTTP/2 priority tree. Extensible priorities does not use stream
+dependencies, which mitigates this vulnerability.
+
+TBD: depending on the outcome of reprioritization discussions, following
+paragraphs may change or be removed.
+
+{{RFC7540}}, Section 5.3.4 describes a scenario where closure of streams in the
+priority tree could cause suboptimal prioritization. To avoid this, {{RFC7540}}
+states that "an endpoint SHOULD retain stream prioritization state for a period
+after streams become closed". Retaining state for streams no longer counted
+towards stream concurrency consumes server resources. Furthermore, {{RFC7540}}
+identifies that reprioritization of a closed stream could affect dependents; it
+recommends updating the priority tree if sufficient state is stored, which will
+also consume server resources. To limit this commitment, it is stated that "The
+amount of prioritization state that is retained MAY be limited" and "If a limit
+is applied, endpoints SHOULD maintain state for at least as many streams as
+allowed by their setting for SETTINGS_MAX_CONCURRENT_STREAMS.". Extensible
+priorities does not use stream dependencies, which minimizes most of the
+resource concerns related to this scenario.
+
+{{RFC7540}}, Section 5.3.4 also presents considerations about the state required
+to store priority information about streams in an "idle" state. This state can
+be limited by adopting the guidance about concurrency limits described above.
+Extensible priorities is subject to a similar consideration because
+PRIORITY_UPDATE frames may arrive before the request that they reference. A
+server could retain the information in order to apply the most up-to-date signal
```

Comment:
Should this "could" be something like "is required to"?

Or maybe, "would be required to ... depending on how the clients use the PRIORITY_UPDATE frames".

Reply:
I think simply "is required to" is the most straightforward thing for now. We can revisit when we circle back on the TODOs after reprioritization chat.